src/main/java/org/truffleruby/core/rope/RopeNodes.java

```diff
@@ -1145,7 +1151,7 @@ public int getCodePointMultiByte(Rope rope, int index,
             final byte[] bytes = bytesNode.execute(rope);
             final Encoding encoding = rope.getEncoding();
 
-            final int characterLength = preciseCharacterLength(encoding, bytes, index, rope.byteLength());
+            final int characterLength = preciseLength(encoding, bytes, index, rope.byteLength());
```

Comment:
Why not create it in `@Cached` here?

Reply:
Refactoring mix-up, I think. I'll make it `@Cached`.

Reply:
Looking more, the reason is because `getCodePointUTF8` calls into `getCodePointMultiByte` and I didn't want to have to `@Cached` on both and pass one to the other.

Reply:
I'll revisit the `mbcToCode` stuff. That can probably be made better for UTF-8.

Reply:
Specializations calling each other is hard for readability, better extract a new function for that case.